### PR TITLE
Support pfm four rows with scientific numbers

### PR DESCRIPTION
**Pull request by ghuls**
Support pfm four rows with scientific numbers.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #...

### CODE REVIEW

File: Bio/motifs/pfm.py [hunk 1]
```diff
@@ -42,8 +42,6 @@ def read(handle, pfm_format):
 def _read_pfm_four_columns(handle):
     """Read motifs in Cluster Buster position frequency matrix format from a file handle.
 
-    Cluster Buster motif format: http://zlab.bu.edu/cluster-buster/help/cis-format.html
```
Comment (peterjc) [inline Bio/motifs/pfm.py:45]: Why remove this URL?

File: Bio/motifs/pfm.py [hunk 2]
```diff
@@ -335,9 +331,9 @@ def _read_pfm_four_rows(handle):
 
     name_pattern = re.compile(r"^>\s*(.+)\s*")
     row_pattern_with_nucleotide_letter = re.compile(
-        r"\s*([ACGT])\s*[[]*[|]*\s*([0-9.\s]+)\s*[]]*\s*"
+        r"\s*([ACGT])\s*[[]*[|]*\s*([0-9.\-eE\s]+)\s*[]]*\s*"
     )
-    row_pattern_without_nucleotide_letter = re.compile(r"\s*([0-9.\s]+)\s*")
+    row_pattern_without_nucleotide_letter = re.compile(r"\s*([0-9.\-eE\s]+)\s*")
```
Comment (peterjc) [inline Bio/motifs/pfm.py:336]: CodeCov reports this line (the second regular expression) is not covered by tests. Do you have a suitable small test case which might be added (using a row without nucleotide).

Comment (ghuls) [reply Bio/motifs/pfm.py:336]: I will try to add a test.

File: Bio/motifs/pfm.py (hunk 1 above)
Comment (peterjc) [reply Bio/motifs/pfm.py:45]: Then the method docstring (line above) needs fixing too?

**Comment by ghuls** [comment]
Recently I found out that Cluster-Buster actually supports floats in the matrix, instead of only integers.

Would it be possible to allow specifying the precision when writing Cluster-Buster files? (0 (integer) and 2 decimal places are actually enough)? 

It would require an interface change in write motifs

```python
write(motifs, fmt, precision=None):
```

Or would there be a better way to do this (introducing 'clusterbuster_int' and 'clusterbuster_float' format for write?

```python
diff --git a/Bio/motifs/clusterbuster.py b/Bio/motifs/clusterbuster.py
index 8340feaf..1572bea1 100644
--- a/Bio/motifs/clusterbuster.py
+++ b/Bio/motifs/clusterbuster.py
@@ -63,7 +63,7 @@ def read(handle):
     return record
 
 
-def write(motifs):
+def write(motifs, precision=0):
     """Return the representation of motifs in Cluster Buster position frequency matrix format."""
     lines = []
     for m in motifs:
@@ -72,7 +72,7 @@ def write(motifs):
         for ACGT_counts in zip(
             m.counts["A"], m.counts["C"], m.counts["G"], m.counts["T"]
         ):
-            lines.append("{:0.0f}\t{:0.0f}\t{:0.0f}\t{:0.0f}\n".format(*ACGT_counts))
+            lines.append("{:0.0f}\t{:0.0f}\t{:0.0f}\t{:0.0f}\n".replace("0f", str(int(precision)) + "f").format(*ACGT_counts))
 
     # Finished; glue the lines together.
     text = "".join(lines)
```

**Review by peterjc — APPROVED** [review]
Looks good to me now, but not my area of expertise. Over to you @mdehoon

**Comment by mdehoon** [comment]
A bit outside of my area of expertise also. I have no objections.

**Review by mdehoon — APPROVED** [review]

**Comment by peterjc** [comment]
CircleCI is stuck, but TravisCI passed so should be fine.

**Comment by peterjc** [comment]
Merged, thank you!

**Pull request merged**